src/poetry/installation/executor.py

```diff
@@ -757,7 +757,7 @@ def _save_url_reference(self, operation: Operation) -> None:
                         writer.writerow([str(path), "", ""])
 
     def _create_git_url_reference(self, package: Package) -> dict[str, Any]:
-        reference = {
+        return {
```

Comment:
this one has gone wrong: we now can never hit the `if package.source_subdirectory:` branch